test/integration/operator_run_bundle_test.go

```diff
@@ -0,0 +1,234 @@
+// repository. The generated csv by default supports OwnNamespace, SingleNamespace and AllNamespaces.
+func createOperator(tc testutils.TestContext, projectName string) {
+	By("initializing a project")
+	err := tc.Init(
+		"--project-version", "3-alpha",
+		"--repo", path.Join("github.com", "example", projectName),
+		"--domain", tc.Domain,
+		"--fetch-deps=false")
+	Expect(err).Should(Succeed())
+
+	By("creating an API definition")
+	err = tc.CreateAPI(
+		"--group", tc.Group,
+		"--version", tc.Version,
+		"--kind", tc.Kind,
+		"--namespaced",
+		"--resource",
+		"--controller",
+		"--make=false")
+	Expect(err).Should(Succeed())
+
+	By("implementing the API")
+	Expect(kbtestutils.InsertCode(
+		filepath.Join(tc.Dir, "api", tc.Version, fmt.Sprintf("%s_types.go", strings.ToLower(tc.Kind))),
+		fmt.Sprintf(`type %sSpec struct {
+`, tc.Kind),
+		`	// +optional
+	Count int `+"`"+`json:"count,omitempty"`+"`"+`
+`)).Should(Succeed())
+
+	By("generating the operator bundle")
+	// Turn off interactive prompts for all generation tasks.
+	replace := "operator-sdk generate kustomize manifests"
+	testutils.ReplaceInFile(filepath.Join(tc.Dir, "Makefile"), replace, replace+" --interactive=false")
+	err = tc.Make("bundle", "IMG="+tc.ImageName)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("building the operator bundle image")
+	// Use the existing image tag but with a "-bundle" suffix.
+	err = tc.Make("bundle-build", "BUNDLE_IMG="+tc.ImageName)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("push the bundle to a remote repository")
+	err = tc.Make("docker-push", "IMG="+tc.ImageName)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func uninstall(kubeconfig string, tc testutils.TestContext) {
+	cfg := &operator.Configuration{KubeconfigPath: kubeconfig}
+	err := cfg.Load()
+	Expect(err).NotTo(HaveOccurred())
+
+	uninstallConfig := operator.NewUninstall(cfg)
+	uninstallConfig.DeleteAll = true
+	uninstallConfig.DeleteOperatorGroupNames = []string{tc.Group}
+	uninstallConfig.Package = defaultOperatorName
+	uninstallConfig.Logf = logrus.Infof
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	err = uninstallConfig.Run(ctx)
+	Expect(err).NotTo(HaveOccurred())
+
+	waitForConfigMapDeletion(ctx, cfg, defaultOperatorName)
+}
+
+func waitForConfigMapDeletion(ctx context.Context, cfg *operator.Configuration, packageName string) {
+	cfgmaps := corev1.ConfigMapList{}
+	opts := []client.ListOption{
+		client.InNamespace(cfg.Namespace),
+		client.MatchingLabels{"owner": "operator-sdk", "package-name": packageName},
+	}
+	err := wait.PollImmediateUntil(250*time.Millisecond, func() (bool, error) {
+		if err := cfg.Client.List(ctx, &cfgmaps, opts...); err != nil {
+			return false, err
+		}
+		return len(cfgmaps.Items) == 0, nil
+	}, ctx.Done())
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func TestRunBundle(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Operator SDK run bundle test suite")
+}
```

Comment:
in order to follow up a standard each dir should have 1 suite test and the tests that will be called by it. IMO we need re-check if is really required to keep this dir.

Reply:
As discussed with @jmrodri and @estroz, this PR was structured to follow the `run packagemanifests` format. We add the generic run package manifest test to operator e2e test which is done here: #3831 and the tests to check various scenarios in the olm integration suite. Similar to how package manifests does: [test/integration/operator_olm_test.go](https://github.com/operator-framework/operator-sdk/blob/master/test/integration/operator_olm_test.go).
I am not sure if these tests would be run when the user calls `run bundle`, because we are using a pre-generated bundle specific for this tests.

Reply:
OK. See that : 
- we cover the features with unit test
- we have e2e tests for each type of project where we will test the run bundle with ALL Namespaces. 

So, this e2e test would just make sense if we would like to cover all possible scenarios. Is this the goal? If yes, OK.
However, note that `test/integration/operator_olm_test.go` is old and is NOT follow the conventions and standards defined so far.  See that: 

- we should use the libs defined and then follow the standards and conventions defined by them and applied in teh ansible/helm tests already. (we have pr to cleanup go implementation waiting for reviews [feat: cleanup go e2e tests and perform improvements](https://github.com/operator-framework/operator-sdk/pull/3499) )
- the integration tests need to be in `test/e2e-<name>`. I suggest we have a dir `test/e2e-olm`
- The directory requires an `e2e_<name>_suite_test` which will have the suite and then the files. Note that it is the standard of the libs adopted. **Create the dir and run the command `$ ginkgo bootstrap` ** then, you will check the files in this way. 
-  we should not re-create the project for each test. the project can be created in the before of the suite and used by all test and the removed on the after of the suite instead. See Ansible/Helm suite tests. 
- the go e2e test should have the [suffix _test which has special meaning for the Go compiler](https://github.com/golang/go/issues/25223)

c/c @jmrodri @estroz 







c/c